File: CS-Examples/03_Cells/SetCellFillPattern.cs
using System;
using System.Windows.Forms;
using Spire.Xls;
using System.Drawing;

namespace SetCellFillPattern
{

	public partial class Form1 : Form
	{
        public Form1()
        {
            InitializeComponent();
        }

		private void btnRun_Click(object sender, EventArgs e)
        {
            // Create a new workbook
            Workbook workbook = new Workbook();

            // Load the document from disk
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\CommonTemplate.xlsx");

            // Get the first worksheet in the workbook
            Worksheet worksheet = workbook.Worksheets[0];

            // Set the cell color for range B7:F7 to yellow
            worksheet.Range["B7:F7"].Style.Color = Color.Yellow;

            // Set the cell fill pattern for range B8:F8 to 125% gray
            worksheet.Range["B8:F8"].Style.FillPattern = ExcelPatternType.Percent125Gray;

            // Save the modified workbook to a file
            string output = "SetCellFillPattern.xlsx";
            workbook.SaveToFile(output, ExcelVersion.Version2013);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            //Launch the file
            ExcelDocViewer(output);
		}
        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }

	}
}

File: CS-Examples/03_Cells/SetCellFillPattern.vb
Imports Spire.Xls

Namespace SetCellFillPattern

	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            'Creates a new instance of the Workbook class.
            Dim workbook As New Workbook()

            'Loads an Excel file from the specified path.
            workbook.LoadFromFile("..\..\..\..\..\..\Data\CommonTemplate.xlsx")
            'Retrieves the first worksheet from the workbook.
            Dim worksheet As Worksheet = workbook.Worksheets(0)

            'Sets the background color of the cells in the range "B7:F7" to yellow.
            worksheet.Range("B7:F7").Style.Color = Color.Yellow

            'Sets the fill pattern of the cells in the range "B8:F8" to a 125% gray pattern.
            worksheet.Range("B8:F8").Style.FillPattern = ExcelPatternType.Percent125Gray

            'Specifies the name of the output file.
            Dim output As String = "SetCellFillPattern.xlsx"
            'Saves the workbook to the specified file in Excel 2013 format.
            workbook.SaveToFile(output, ExcelVersion.Version2013)
            ' Release the resources used by the workbook
            workbook.Dispose()

            'Launch the file
            ExcelDocViewer(output)
		End Sub
		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub

	End Class
End Namespace
